install/powershellscript/README.md

```diff
@@ -0,0 +1,112 @@
+
+It installs the Microsoft Go toolset into a directory of your choice, or defaults to a directory in the user-specific data directory.
+See `microsoft-go-install.ps1 -h` for more information about the parameters and defaults.
+
+The script is intended for use in CI/CD pipelines or to reproduce the results of those CI/CD pipelines locally.
+
+## Prerequisites
+
+On non-Windows platforms, install [PowerShell (`pwsh`)](https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell).
+
+On Windows, either Windows PowerShell or PowerShell can be used.
+
+> [!NOTE]
+> PowerShell was formerly known as PowerShell Core.
+> Now [Windows PowerShell and PowerShell](https://learn.microsoft.com/en-us/powershell/scripting/what-is-windows-powershell) are the names used by Microsoft for these products.
+
+## Usage
+
+Run the script using the `pwsh` command:
+
+```bash
+pwsh ./microsoft-go-install.ps1
+```
+
+If you use a PowerShell terminal, you can also choose to run the script directly:
+
+```
+.\microsoft-go-install.ps1
+```
+
+Running the script directly allows the script to change the terminal's `PATH` so the installed Go binary is then available in the current session as `go`.
+
+Note that in typical CI/CD pipelines, each step is run in a fresh process and the `PATH` change will not be preserved.
+If you're using Azure Pipelines, see the help message for `-AzurePipelinePath`.
+
+Pass `-h` to show help.
+
+## Where to put the script
+
+The script can be placed in the root of your repository or in a subdirectory.
+It can be run from a different directory with no effect on the results.
+
+To copy the script and keep it up to date in a reproducible way, use the `microsoftgoinstallscript` command.
+
+### Set up `microsoftgoinstallscript`
+
+In your Go module, run:
+
+```
+go get github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript@latest
+go install github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript
+```
+
+If you don't already use `github.com/microsoft/go-infra` in your module, add a `tools/tools.go` file to your module to pin the version of `microsoftgoinstallscript` and prevent `go mod tidy` from removing it:
+
+```go
+//go:build tools
+
+package tools
+
+import (
+    _ "github.com/microsoft/go-infra/install/powershellscript"
+)
+```
+
+> [!NOTE]
+> This is a well-known workaround to pin the version of a tool in a Go module.
+> See the [Go wiki](https://go.dev/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) for more information.
+
+Then, in the directory of your choice inside your module, run:
+
+```
+microsoftgoinstallscript
+```
+
+### Updating the script using `microsoftgoinstallscript`
+
+To update the script, run this in the directory where the script is stored:
+
+```
+go get github.com/microsoft/go-infra/install/powershellscript/cmd/microsoftgoinstallscript@latest
```

Comment:
The package name is too verbose for my old eyes. Users will mainly want to install the CLI rather than using the  small Go API in `github.com/microsoft/go-infra/install/powershellscript`. When the principal "product" is a CLI and not the Go API, it is common to not follow the usual `pkgname/cmd/cliname` convention but use `cliname/pkgname` instead. This way the CLI is easier to install (requires less typing!).

Also, the `microsoftgoinstallscript` cli can potentially be made more generic and install other script types. Even if we don't eventually do it, I don't see a lot of value on putting it under a folder named `powershellscript`, as we are artificially limiting the scope of the cli.

With all this in mind, this is my suggestion:

- Rename `github.com/microsoft/go-infra/install/powershellscript` to `github.com/microsoft/go-infra/goinstallscript`.
- Move all files in `github.com/microsoft/go-infra/goinstallscript` to `github.com/microsoft/go-infra/goinstallscript/powershell`.
- Move `github.com/microsoft/go-infra/goinstallscript/cmd/microsoftgoinstallscript/main.go` to `github.com/microsoft/go-infra/goinstallscript`.

If we do this, then installing the cli will look like this:

```
go get github.com/microsoft/go-infra/goinstallscript@latest
```